mirrord/agent/src/runtime.rs

```diff
@@ -131,23 +131,51 @@ pub(crate) struct ContainerdContainer {
     container_id: String,
 }
 
+/// Connects to the given containerd socket
+/// and returns the client only if the given container
+/// exists.
+async fn connect_and_find_container(
+    container_id: String,
+    sock_path: impl AsRef<std::path::Path>,
+) -> Result<TasksClient<Channel>> {
+    let channel = connect(sock_path).await?;
+    let mut client = TasksClient::new(channel);
+    let request = GetRequest {
+        container_id,
+        ..Default::default()
+    };
+    let request = with_namespace!(request, DEFAULT_CONTAINERD_NAMESPACE);
+    client.get(request).await?;
```

Comment:
Will this error always? or should also check `.process.is_some().ok_or(AgentError...)?`